File: plant_nursery_management_system/src/main/java/com/project/plant_nursery_management_system/service/serviceInterface/OrderService.java
package com.project.plant_nursery_management_system.service.serviceInterface;

import com.project.plant_nursery_management_system.dto.OrderDto;
import com.project.plant_nursery_management_system.model.Order;
import com.project.plant_nursery_management_system.model.onlyenums.OrderStatus;

import java.util.List;


public interface OrderService {
    Order placeOrder(OrderDto orderDTO);
    Order getOrderById(Long id);
    List<Order> getOrdersByUserId(Long userId);
    List<Order> getAllOrders();

    Order cancelOrder(Long orderId);
    Order updateOrder(Long orderId, OrderDto updatedOrderDTO);
    void deleteOrder(Long orderId , Long userId, boolean isAdmin);

    Order updateOrderStatus(Long orderId, OrderStatus status);
}

File: PNMS/plant-nusery/src/services/admin/orderService.js
import axiosInstance from "../axiosInstance";

export const getAllOrders = async () => {
  return axiosInstance.get("/orders");
};

export const deleteOrder = async (id) => {
  return axiosInstance.delete(`/orders/${id}`);
};

export const updateOrder = async (id, updatedData) => {
  return axiosInstance.put(`/orders/status/${id}`, updatedData);
};
